yamllint/rules/hyphens.py

```diff
@@ -72,24 +87,97 @@
     -    key
     -   key2
     -  key42
+
+#. With ``hyphens: {min-spaces-after: 3}``
+
+   the following code snippet would **PASS**:
+   ::
+
+    list:
+    -   key
+    -    key2
+    -     key42
+    -foo:  # starter of a new sequence named "-foo";
+           # without the colon, a syntax error will be raised.
+
+   the following code snippet would **FAIL**:
+   ::
+
+    -  key
+    -   key2
+    -  key42
+
+#. With ``hyphens: {min-spaces-after: 3, check-scalars: true}``
+
+   the following code snippet would **PASS**:
+   ::
+
+    list:
+    -   key
+    -    key2
+    -     key42
+    key: -value
+
+   the following code snippets would **FAIL**:
+   ::
+
+    ---
+    -item0
+
+   ::
+
+    sequence:
+      -key  # Mind the spaces before the hyphen to enforce
+            # the sequence and avoid a syntax error
 """
 
 
 import yaml
 
+from yamllint.linter import LintProblem
 from yamllint.rules.common import spaces_after
 
 
 ID = 'hyphens'
 TYPE = 'token'
-CONF = {'max-spaces-after': int}
-DEFAULT = {'max-spaces-after': 1}
+CONF = {'max-spaces-after': int,
+        'min-spaces-after': int,
+        'check-scalars': bool}
+DEFAULT = {'max-spaces-after': 1,
+           'min-spaces-after': -1,
+           'check-scalars': False}
+
+
+def VALIDATE(conf):
+    if conf['max-spaces-after'] == 0:
+        return '"max-spaces-after" cannot be set to 0'
+    if (conf['min-spaces-after'] > 0 and
+            conf['min-spaces-after'] > conf['max-spaces-after']):
+        return '"min-spaces-after" cannot be greater than "max-spaces-after"'
 
 
 def check(conf, token, prev, next, nextnext, context):
     if isinstance(token, yaml.BlockEntryToken):
-        problem = spaces_after(token, prev, next,
-                               max=conf['max-spaces-after'],
-                               max_desc='too many spaces after hyphen')
-        if problem is not None:
-            yield problem
+        if conf['max-spaces-after'] > 0:
+            problem = spaces_after(token, prev, next,
+                                   max=conf['max-spaces-after'],
+                                   max_desc='too many spaces after hyphen')
+            if problem is not None:
+                yield problem
+
+        if conf['min-spaces-after'] > 0:
```

Comment:
Doing the check if `conf['min-spaces-after'] == 1` is useless (it cannot be lower than 1) and will degrade performance.

I propose:
```suggestion
        if conf['min-spaces-after'] > 1:
```